pcsx2/GS/Renderers/Metal/GSDeviceMTL.mm

```diff
@@ -1475,8 +1475,20 @@ static void OnMainThread(Fn&& fn)
 
 	GSTextureMTL* sT = static_cast<GSTextureMTL*>(sTex);
 	GSTextureMTL* dT = static_cast<GSTextureMTL*>(dTex);
+	const GSVector4i dst_rect(0, 0, dT->GetWidth(), dT->GetHeight());
+	const bool full_draw_copy = dst_rect.eq(r);
 
-	// Process clears
+	// Source is cleared, if destination is a render target, we can carry the clear forward.
+	if (sT->GetState() == GSTexture::State::Cleared && dT->IsRenderTargetOrDepthStencil())
+	{
+		if (ProcessClearsBeforeCopy(sTex, dTex, full_draw_copy))
+			return;
```

Comment:
```suggestion
	if (sT->GetState() == GSTexture::State::Cleared)
	{
		if (dT->IsRenderTargetOrDepthStencil() && ProcessClearsBeforeCopy(sTex, dTex, full_draw_copy))
			return;
```
We should probably flush clears on src if dst is not a render target or depth/stencil

Reply:
Yep, you're right.